fuse/eval/evaluator.py

```diff
@@ -110,7 +110,9 @@ def eval(
                     raise
         elif batch_size > 0:
             # data is iterable - each iteration is a single sample (represented by a dictionary)
-            assert isinstance(data, Iterable), "Error: batch mode (batch_size != None) supports only iterable data"
+            assert isinstance(data, Iterable) and not isinstance(
+                data, pd.DataFrame
+            ), "Error: batch mode (batch_size != None) supports only iterable data"
```

Comment:
minor, but maybe worth writing:

"Error: batch mode (batch_size != None) supports only iterable data and doesn't support pandas' DataFrame"